src/curve_fitting/canvas.cljs

```diff
@@ -5,6 +5,52 @@
 
 (enable-console-print!)
 
+(defn draw-x-axis [graph]
+  (let [ctx (.getContext (:canvas graph) "2d")
+        width (* (:scale-x graph) (:range-x graph))
+        center-y (:center-y graph)]
+    (set! (.-globalAlpha ctx) 1.0)
+    (.beginPath ctx)
+    (.moveTo ctx 0 center-y)
+    (.lineTo ctx width center-y)
+    (set! (.-strokeStyle ctx) "black")
+    (.stroke ctx)
+    ctx))
+
+(defn draw-y-axis [graph]
+  (let [ctx (.getContext (:canvas graph) "2d")
+        height (* (:scale-y graph) (:range-y graph))
+        center-x (:center-x graph)]
+    (set! (.-globalAlpha ctx) 1.0)
+    (.beginPath ctx)
+    (.moveTo ctx center-x 0)
+    (.lineTo ctx center-x height)
+    (set! (.-strokeStyle ctx) "black")
+    (.stroke ctx)
+    ctx))
+
+(defn update-x-axis [graph]
+  (let [ctx (.getContext (:canvas graph) "2d")]
+    (set! (.-globalAlpha ctx) 1.0)
+    (.beginPath ctx)
+    (.moveTo ctx (:min-x graph) 0)
+    (.lineTo ctx (:max-x graph) 0)
+    (set! (.-strokeStyle ctx) "black")
+    (set! (.-lineWidth ctx) (/ 1 (:scale-x graph)))
+    (.stroke ctx)
+    ctx))
+
+(defn update-y-axis [graph]
+  (let [ctx (.getContext (:canvas graph) "2d")]
+    (set! (.-globalAlpha ctx) 1.0)
+    (.beginPath ctx)
+    (.moveTo ctx 0 (:min-y graph))
+    (.lineTo ctx 0 (:max-y graph))
+    (set! (.-strokeStyle ctx) "black")
+    (set! (.-lineWidth ctx) (/ 1 (:scale-x graph)))
+    (.stroke ctx)
+    ctx))
```

Comment:
There's a lot of duplication between all three of these functions. Might be worth exploring attempting to DRY them out.